File: server/bak/ncurses/dialog.php
<?php

class Dialog extends Widget {

    function __construct($text, $x, $y, $width, $height) {
        parent::__construct($x, $y, $width, $height, $this);
        $this->text = $text;
        $this->x = $x;
        $this->y = $y;
        $this->width = $width;
        $this->height = $height;
    }

    function __destruct() {
    }

    function draw($screen) {
        parent::draw($screen);
        // draw frame using characters
        for ($x = 0; $x < $this->width; $x++) {
            $screen->move($this->x + $x, $this->y);
            $screen->printw('=');
            $screen->move($this->x + $x, $this->y + $this->height-1);
            $screen->printw('-');
        }
        for ($y = 0; $y < $this->height; $y++) {
            $screen->move($this->x, $this->y + $y);
            $screen->printw('|');
            $screen->move($this->x + $this->width-1, $this->y + $y);
            $screen->printw('|');
        }       
        $screen->move($this->x + 1, $this->y);
        $screen->printw($this->text);
     }

}

?>

File: server/bak/ncurses/dialog.py
from ncurses.widget import Widget

class Dialog(Widget):

    def __init__(self, text, x, y, width, height):
        super().__init__(x, y, width, height, self)
        self.text = text
        self.x = x
        self.y = y
        self.width = width
        self.height = height


    def draw(self, screen):
        super().draw(screen)
        # draw frame using characters
        for x in range(0, self.width):
            screen.move(self.x + x, self.y)
            screen.printw('=')
            screen.move(self.x + x, self.y + self.height-1)
            screen.printw('-')
        
        for y in range(0, self.height):
            screen.move(self.x, self.y + y)
            screen.printw('|')
            screen.move(self.x + self.width-1, self.y + y)
            screen.printw('|')
            
        screen.move(self.x + 1, self.y)
        screen.printw(self.text)
